include/fastrtps/utils/DBQueue.h

```diff
@@ -77,6 +84,22 @@ class DBQueue {
       mForegroundQueue->pop();
    }
 
+   //! Return the front element in the foreground queue by moving it and erase it from the queue.
+   T FrontAndPop() const
```

Comment:
```suggestion
   T FrontAndPop()
```

Reply:
It is a mistery for me why it worked. 
Probably because pop is done from a ptr, and this the actual object (ptr) has not change